src/main/java/uk/gov/cshr/civilservant/service/OrganisationalUnitService.java

```diff
@@ -179,4 +187,25 @@ public OrganisationalUnit deleteAgencyToken(OrganisationalUnit organisationalUni
     public Optional<OrganisationalUnit> get(Long id) {
         return repository.findById(id);
     }
+
+    public List<String> getOrganisationalUnitCodesForIds(List<String> organisationIds) {
+        return repository.findAllCodesForIds(organisationIds);
+    }
+
+    public List<Long> getOrganisationIdWithChildrenIds(List<String> listOrganisationCodes) {
+        List<Long> listOrganisationIds = new ArrayList<>();
+        for (String code : listOrganisationCodes) {
+            getOrganisationWithChildren(code).forEach(x -> listOrganisationIds.add(x.getId()));
+        }
+        return listOrganisationIds;
+    }
+
+    public void addOrganisationReportingPermission(Long id, List<Long> organisationIds) {
+        List<CivilServantOrganisationReportingPermission> list = new ArrayList<>();
+        for (Long orgId : organisationIds) {
+            CivilServantOrganisationReportingPermission entity = new CivilServantOrganisationReportingPermission(id, orgId);
+            list.add(entity);
+        }
```

Comment:
Above code could be implemented using Java 8 Stream APIs.

Reply:
same reason as above.

Reply:
As discussed further over the phone, this code should be written using stream.